src/core/ext/filters/client_channel/lb_policy_factory.h

```diff
@@ -66,9 +66,18 @@ typedef struct grpc_lb_addresses {
 grpc_lb_addresses *grpc_lb_addresses_create(
     size_t num_addresses, const grpc_lb_user_data_vtable *user_data_vtable);
 
+/** Deep copy the address from \a src to \a dest. */
+void grpc_lb_address_copy(grpc_lb_address* dest, const grpc_lb_address* src,  
+                          const grpc_lb_user_data_vtable* user_data_vtable);
+
 /** Creates a copy of \a addresses. */
 grpc_lb_addresses *grpc_lb_addresses_copy(const grpc_lb_addresses *addresses);
 
+/** Extract the backend addresses and return them in a new \a 
+    grpc_lb_addresses. */
+grpc_lb_addresses *grpc_lb_addresses_extract_backends(
```

Comment:
You should definitely not change `grpc_arg_pointer_vtable`.  But you should not need to do so.

The goal of the channel arg pointer vtable is to provide hooks for copying an arbitrary object encoded in a channel arg as a pointer.  If we're copying a channel arg, we don't want to change its value; we just want to copy it as-is.  So I think you can just change the copy function that we're using for this particular channel arg to pass in the parameter that causes us to copy all addresses, not just the backend addresses.